autoload/vista/highlight.vim

```diff
@@ -24,8 +24,9 @@ function! vista#highlight#Add(lnum, ensure_visible, tag) abort
     " If we know the tag, then what we have to do is to use the length of tag
     " based on the starting point.
     "
-    " start is 0-based, while the column used in matchstrpos is 1-based.
+    " start is 0-based, while the column used in stridx or matchstrpos is 1-based.
     if !empty(a:tag)
+      let start = stridx(cur_line, a:tag)
```

Comment:
Which case does this line fix?

Reply:
> Which case does this line fix?

A tag name string may start with `~`, the previous `start` will skip `~`, causing the highlighting area to move backward.
And I am not quite clear about the syntax highlighting on `vista.vim`, may not be able to do more fixes now.

Reply:
Then there is a conflict here as `~` is also used for the symbol of Protected.  This should be resolved in another PR, please don't fix in this one.